lib/compat/wordpress-6.3/script-loader.php

```diff
@@ -13,3 +13,73 @@
 remove_action( 'wp_body_open', 'wp_global_styles_render_svg_filters' );
 remove_action( 'in_admin_header', 'wp_global_styles_render_svg_filters' );
 
+/**
+ * Collect the block editor assets that need to be loaded into the editor's iframe.
+ *
+ * @since 6.0.0
+ * @access private
+ *
+ * @return array {
+ *     The block editor assets.
+ *
+ *     @type string|false $styles  String containing the HTML for styles.
+ *     @type string|false $scripts String containing the HTML for scripts.
+ * }
+ */
+function _gutenberg_get_iframed_editor_assets() {
+	global $wp_styles, $wp_scripts;
+
+	// Keep track of the styles and scripts instance to restore later.
+	$current_wp_styles  = $wp_styles;
+	$current_wp_scripts = $wp_scripts;
+
+	// Create new instances to collect the assets.
+	$wp_styles  = new WP_Styles();
+	$wp_scripts = new WP_Scripts();
+
+	// Register all currently registered styles and scripts. The actions that
+	// follow enqueue assets, but don't necessarily register them.
+	$wp_styles->registered  = $current_wp_styles->registered;
+	$wp_scripts->registered = $current_wp_scripts->registered;
+
+	wp_enqueue_style( 'wp-block-editor-content' );
+	// To do: investigate why this is not enqueued through enqueue_block_assets,
+	// as styles for non-core blocks are.
+	wp_enqueue_style( 'wp-block-library' );
+	wp_enqueue_script( 'wp-polyfill' );
+
+	// We don't want to load EDITOR scripts and styles in the iframe, only
+	// assets for the content.
+	add_filter( 'should_load_block_editor_scripts_and_styles', '__return_false' );
+	do_action( 'enqueue_block_assets' );
+	remove_filter( 'should_load_block_editor_scripts_and_styles', '__return_false' );
+
+	ob_start();
+	wp_print_styles();
+	wp_print_fonts();
```

Comment:
There is an issue using `wp_print_fonts` since it will [do nothing if the handles are empty](https://github.com/WordPress/gutenberg/blob/d54dc62685dd1b9980c4a06d3f488b87b6c7fe10/lib/experimental/fonts-api/class-wp-fonts.php#L453).

Before, [Gutenberg assigned the registered fonts to the queue](https://github.com/WordPress/gutenberg/blob/d54dc62685dd1b9980c4a06d3f488b87b6c7fe10/lib/compat/wordpress-6.2/script-loader.php#L140) so `$wp_fonts` can fallback to the handles in the queue. However, this function doesn't implement that mechanism and it leads the font styles won't be printed

Reply:
@hellofromtonya Is this fixed now since [[Fonts API] Relocate which fonts to print from script-loader into wp_print_fonts()](https://github.com/WordPress/gutenberg/pull/49655#top)? Thanks for looking into it!

Reply:
>Is this fixed now since https://github.com/WordPress/gutenberg/pull/49655#top? Thanks for looking into it!

@ellatrix Yes, it is fixed via PR #49655. That change was cherry-picked for 15.7.1, which will be released today.